optuna/samplers/_nsga2/sampler.py

```diff
@@ -140,7 +146,9 @@ def reseed_rng(self) -> None:
     def infer_relative_search_space(
         self, study: Study, trial: FrozenTrial
     ) -> Dict[str, BaseDistribution]:
-        return {}
+
+        search_space = intersection_search_space(study)
```

Comment:
It's worth highlighting that this change in introducing the intersection search space, while it aligns this sampler with other relative samplers, it's also introduces a change in behavior for conditional parameters. How about leaving a note about that the PR description, making the change easier to track in the future.